eth_client/lib/eth_client/contract/opcodes.ex

```diff
@@ -0,0 +1,70 @@
+defmodule EthClient.Contract.Opcodes do
+  alias EthClient.Rpc
+  # Temporary storage for the opcodes,
+  # should I improve this?
+  @opcodes "./opcodes.json"
+           |> Path.expand()
+           |> File.read!()
+           |> Jason.decode!()
+           |> Enum.reduce(%{}, fn %{"Hex" => hex, "Name" => name}, map_acum ->
+             case name do
+               "*invalid*" -> map_acum
+               _ -> Map.put(map_acum, hex, name)
+             end
+           end)
```

Comment:
This must go in a function. I don't see any advantage to having it in a module variable.

```elixir
def get_opcodes() do
   opcodes_from_file!()
   |> parse_opcodes()
end

def opcodes_from_file!() do
        "./opcodes.json"
        |> Path.expand()
        |> File.read!()
end

def parse_opcodes(codes) do
   codes
   |> Jason.decode!()
   |> filter_invalid()
end

def filter_invalid(code_list) do
  Enum.reduce(code_list, fn 
      (%{"Hex" => _hex, "Name" => name}, acc) when name == "*invalid*" -> acc
      (%{"Hex" => hex, "Name" => name}, acc) -> Map.put(acc, hex, name)
  end)
end
```
If you still want to use the variable:
```elixir
@opcodes get_opcodes()
```

**Advantages in these changes:**

* It is more readable
* If an error occurs, the compiler will tell you in which function and you save time to identify it
* In the future you can use some functions and avoid reworking

Reply:
Thanks for the feedback!


My rationale for using `@opcodes` was that, since module attributes are evaluated on compile time, we would have the opcodes map ready when using the module from iex. But your approach looks a lot cleaner and more easy to debug, so I've added it to the PR.

Reply:
If you want to have functions available in `iex` I suggest you create a module inside the `.iex.exs` file:

```elixir
# .iex.exs file

# Import ecto queries for testing queries in iex
import_if_available(Ecto.Query)

#  Application.load(:tzdata)
#  :ok = Application.ensure_started(:tzdata)
opcodes_file = "~/Workspace/project/opcodes.json"

defmodule Local do
  def load_opcodes() do
        opcodes_file
        |> Path.expand()
        |> File.read!()
  end
end
```
Now you can load the codes in memory in your local env to perform tests, you just call the function that will be available only in your local but that can be used by any dev when downloading the project.

```elixir
iex(1)> opcodes = Local.load_opcodes()
```